devito/dle/backends/yask.py

```diff
@@ -1,27 +1,190 @@
+    suitable for the YASK storage layout.
+
+    Subclassing ``numpy.ndarray`` would have led to shadow data copies, because
+    of the different storage layout.
+    """
+
+    # Force __rOP__ methods (OP={add,mul,...) to get arrays, not scalars, for efficiency
+    __array_priority__ = 1000
+
+    def __new__(cls, name, shape, dimensions, dtype, buffer=None):
+        """
+        Create a new YASK Grid and attach it to a "fake" solution.
+        """
+        # Init YASK if not initialized already
+        init(dimensions, shape, dtype)
+        # Only create a YaskGrid if the requested grid is a dense one
+        if tuple(i.name for i in dimensions) == YASK.dimensions:
+            obj = super(YaskGrid, cls).__new__(cls)
+            obj.__init__(name, shape, dimensions, dtype, buffer)
+            return obj
+        else:
+            return None
+
+    def __init__(self, name, shape, dimensions, dtype, buffer=None):
+        self.name = name
+        self.shape = shape
+        self.dimensions = dimensions
+        self.dtype = dtype
+
+        self.grid = YASK.setdefault(name)
 
-try:
-    import yask_compiler as yask
-    # Factories to interact with YASK
-    cfac = yask.yask_compiler_factory()
-    fac = yask.node_factory()
-except ImportError:
-    yask = None
+        # Always init the grid, at least with 0.0
+        self[:] = 0.0 if buffer is None else buffer
+
+    def __getitem__(self, index):
+        # TODO: ATM, no MPI support.
+        start, stop, shape = convert_multislice(as_tuple(index), self.shape)
+        if not shape:
+            debug("YaskGrid: Getting single entry")
+            assert start == stop
+            out = self.grid.get_element(*start)
+        else:
+            debug("YaskGrid: Getting full-array/block via index [%s]" % str(index))
+            out = np.empty(shape, self.dtype, 'C')
+            self.grid.get_elements_in_slice(out.data, start, stop)
+        return out
+
+    def __setitem__(self, index, val):
+        # TODO: ATM, no MPI support.
+        start, stop, shape = convert_multislice(as_tuple(index), self.shape, 'set')
+        if all(i == 1 for i in shape):
+            debug("YaskGrid: Setting single entry")
+            assert start == stop
+            self.grid.set_element(val, *start)
+        elif isinstance(val, np.ndarray):
+            debug("YaskGrid: Setting full-array/block via index [%s]" % str(index))
+            self.grid.set_elements_in_slice(val, start, stop)
+        elif all(i == j-1 for i, j in zip(shape, self.shape)):
+            debug("YaskGrid: Setting full-array to given scalar via single grid sweep")
+            self.grid.set_all_elements_same(val)
+        else:
+            debug("YaskGrid: Setting block to given scalar via index [%s]" % str(index))
+            self.grid.set_elements_in_slice_same(val, start, stop)
+
+    def __getslice__(self, start, stop):
+        if stop == sys.maxint:
```

Comment:
I found it in some "official docs", which now I can't find again, argh. 
Here's a pointer though : https://stackoverflow.com/questions/25597815/slice-endpoints-invisibly-truncated

In practice, this is something you need to do to match NumPy's behaviour. I should have added a comment